File: chapter_003/struct_bmi.py
#!/bin/bash/env python3

height_cm = float(input("身長(cm): "))
weight_kg = float(input("体重(kg): "))

height_m = height_cm / 100.0
bmi = weight_kg / (height_m ** 2)

bmi_list = [
    {"min": 0, "max": 18.5, "level": "痩せ型"},
    {"min": 18.5, "max": 25, "level": "標準型"},
    {"min": 25, "max": 30, "level": "肥満（1度）"},
    {"min": 30, "max": 35, "level": "肥満（2度）"},
    {"min": 35, "max": 40, "level": "肥満（3度）"},
    {"min": 40, "max": "", "level": "肥満（4度）"},
]

result = ""
for range in bmi_list:
    if range["min"] <= bmi < range["max"]:
        result = range["level"]

print(f"BMI: {bmi:.2f} {result}")

File: chapter_003/struct_bmi.rs
struct BmiRange {
    min: f64,
    max: f64,
    lavel: &'static str,
}

fn bmi_calc(height: f64, weight: f64) -> f64 {
    weight / (height / 100.0).powf(2.0)
}

fn user_input(prompt: &str) -> f64 {
    println!("{}", prompt);
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Failed to read line");
    input.trim().parse::<f64>().expect("Failed to parse input")
}

fn main() {
    let height = user_input("Enter your height (cm): ");
    let weight = user_input("Enter your weight (kg): ");
    let bmi = bmi_calc(height, weight);

    let bmi_ranges = vec![
    BmiRange {
        min: 18.5,
        max: 25.0,
        lavel: "Normal",
    },
    BmiRange {
        min: 25.0,
        max: 30.0,
        lavel: "Overweight",
    },
    BmiRange {
        min: 30.0,
        max: 35.0,
        lavel: "Obesity",
    },
    BmiRange {
        min: 35.0,
        max: 40.0,
        lavel: "Severe obesity",
    },
];

    println!("Your BMI is {:.2}", bmi);
    for range in bmi_ranges {
        if bmi >= range.min && bmi <= range.max {
            println!("You are in the {} range.", range.lavel);
        }
    }
}
